src/runtime_src/tools/xclbinutil/XclBinClass.cxx

```diff
@@ -1585,62 +1605,62 @@ XclBin::setKeyValue(const std::string & _keyValue)
   throw std::runtime_error(errMsg);
 }
 
-void 
-XclBin::removeKey(const std::string & _sKey)
+void
+XclBin::removeKey(const std::string& _sKey)
 {
 
   XUtil::TRACE(XUtil::format("Removing User Key: '%s'", _sKey.c_str()));
 
-  Section *pSection = findSection(KEYVALUE_METADATA);
+  Section* pSection = findSection(KEYVALUE_METADATA);
   if (pSection == nullptr) {
     std::string errMsg = XUtil::format("ERROR: Key '%s' not found.", _sKey.c_str());
     throw std::runtime_error(errMsg);
   }
 
   boost::property_tree::ptree ptKeyValueMetadata;
-   pSection->getPayload(ptKeyValueMetadata);
-
-   XUtil::TRACE_PrintTree("KEYVALUE:", ptKeyValueMetadata);
-   boost::property_tree::ptree ptKeyValues = ptKeyValueMetadata.get_child("keyvalue_metadata");
-   std::vector<boost::property_tree::ptree> keyValues = XUtil::as_vector<boost::property_tree::ptree>(ptKeyValues, "key_values");
-
-   // Update existing key
-   bool bKeyFound = false;
-   for (unsigned int index = 0; index < keyValues.size(); ++index) {
-      if (keyValues[index].get<std::string>("key") == _sKey) {
-         bKeyFound = true;
-         XUtil::QUIET(std::string("Removing key '") + _sKey + "'");
-         keyValues.erase(keyValues.begin() + index);
-         break;
-      }
+  pSection->getPayload(ptKeyValueMetadata);
+
+  XUtil::TRACE_PrintTree("KEYVALUE:", ptKeyValueMetadata);
+  boost::property_tree::ptree ptKeyValues = ptKeyValueMetadata.get_child("keyvalue_metadata");
+  std::vector<boost::property_tree::ptree> keyValues = XUtil::as_vector<boost::property_tree::ptree>(ptKeyValues, "key_values");
+
+  // Update existing key
+  bool bKeyFound = false;
+  for (unsigned int index = 0; index < keyValues.size(); ++index) {
+    if (keyValues[index].get<std::string>("key") == _sKey) {
+      bKeyFound = true;
+      XUtil::QUIET(std::string("Removing key '") + _sKey + "'");
+      keyValues.erase(keyValues.begin() + index);
+      break;
     }
+  }
 
-   if (bKeyFound == false) {
-     std::string errMsg = XUtil::format("ERROR: Key '%s' not found.", _sKey.c_str());
-     throw std::runtime_error(errMsg);
-   }
+  if (bKeyFound == false) {
+    std::string errMsg = XUtil::format("ERROR: Key '%s' not found.", _sKey.c_str());
+    throw std::runtime_error(errMsg);
+  }
 
-   // Now create a new tree to add back into the section
-   boost::property_tree::ptree ptKeyValuesNew;
-   for (auto keyvalue : keyValues) {
-     ptKeyValuesNew.push_back(std::make_pair("", keyvalue));
-   }
+  // Now create a new tree to add back into the section
+  boost::property_tree::ptree ptKeyValuesNew;
+  for (auto keyvalue : keyValues) {
```

Comment:
auto& ?

Reply:
Fixed here and in other locations of this file.